psi4/src/psi4/libmints/mintshelper.h

```diff
@@ -297,6 +297,8 @@ class PSI_API MintsHelper {
     std::vector<SharedMatrix> ao_quadrupole();
     /// Vector AO Traceless Quadrupole Integrals
     std::vector<SharedMatrix> ao_traceless_quadrupole();
+    /// Vector AO Multipole Integrals up to given order
+    std::vector<SharedMatrix> ao_multipoles(const std::vector<double>& origin = {0., 0., 0.}, int order = 1);
```

Comment:
I like this general `ao_multipoles` function a lot! Could we get a comment here explaining the order that the multipoles are returned in? Would it be [x, y, z, xx, xy, xz, yy, yz, zz, xxx, ...]? You might've already written that comment somewhere else.

Reply:
I like the general function, too. However, `order` shouldn't have a default value; the caller should explicitly specify it.

Reply:
I'm not sure about that, I wanted to keep the order as in similar `mintshelper` functions.

Reply:
Ah, I see. Yeah, I would prefer the parameters on the other functions have the multipole order first. I personally think a caller to a multipoles function should provide the order they want but not necessarily the origin.

Reply:
Perhaps file an issue to change the argument order for the other functions?

Reply:
Yeah, that sounds good @JonathonMisiewicz .

Reply:
Ok, so we change the order for the multipole function already here, rest comes later?

Reply:
Yes. I just created an issue to change the other.

Reply:
Does having a default origin make sense, for arbitrary order multipoles?  The moments are origin dependent from `n+1`, upwards, where `n` is the order of the first nonzero moment.  So, e.g., the dipole is center-dependent in charged molecules but not in neutral molecules.  If we're talking about allowing high orders in this function, the majority of them are going to be origin-dependent.  Max already mentioned external potentials, but for things like CFMM one must also specify the expansion center (before then translating that using fancy operators).  That's all a verbose way of me trying to say that I don't think either the origin or the expansion order should be given default arguments here.

Reply:
I'm all for *no* default values for the generalized functions (should be changed to `ao_multipole_potential`, too, I can do that when I add M-D for these integrals). If one is not aware that the moments are indeed origin-dependent, better throw an error than returning a probably unintended result IMHO.
For the specialized functions that are exposed via `MintsHelper` there is not even the possibility to change the origin, but this is fine for now I guess.